src/main/java/client/Character.java

```diff
@@ -3107,11 +3110,32 @@ private synchronized void gainExpInternal(long gain, int equip, int party, boole
             if (leftover > 0) {
                 gainExpInternal(leftover, equip, party, false, inChat, white);
             } else {
+                saveExpLogToDB();
```

Comment:
This doesn't seem like a good idea... saving to db on every exp gained will result in bad performance. 
What you could do is push it to some queue and process it by some other thread to save it asynchronously. If you're willing to go that far for this feature...